File: module_6/thi-module6/src/main/java/com/example/demo/model/product/ProductType.java
package com.example.demo.model.product;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Setter
@Getter
@RequiredArgsConstructor
public class ProductType {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private Double price;
    private String author;
    @Column(columnDefinition = "Date")
    private Date creationDate;
    private String category;
    private String careSugarOrSalty;
    private String madeIn;
    private String yesNoGas;

    @ManyToOne
    @JoinColumn(name = "id_rent_type", referencedColumnName = "id")
    private RentType rentType;

    @JsonBackReference
    @OneToMany(mappedBy = "productType")
    private List<Product> productList;
}

File: module_6/sprint-angular/src/app/model/product-type.ts
import {Drink} from "./drink";
import {Book} from "./book";
import {Food} from "./Food";
import {RentType} from "./rentType";

export interface ProductType {
  id?: number,
  name?: string
  rentType?: RentType,
  price?:number,
  author?: string,
  creationDate?: string,
  category?: string,
  madeIn?: string,
  yesNoGas?: string,
  careSugarOrSalty?: string,
}
